### PR TITLE
Add Api.hpp

### PR DESCRIPTION
**Pull request by guillaumeAbel**

### CODE REVIEW

**Comment by guillaumeAbel** [comment]
Miss ISceneManagerType

File: include/Api.hpp [hunk 1]
```diff
@@ -0,0 +1,55 @@
+/*
+** EPITECH PROJECT, 2023
+** Visual Studio Live Share (Workspace)
+** File description:
+** ICore
+*/
+
```
Comment (Saverio976) [inline include/Api.hpp:7]: add ifndef for windows with EXPORT like this: https://github.com/Touxooo/archi-arcade-toulouse/blob/main/include/api.h

File: include/Api.hpp [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+#pragma once
+
+#include "ISystemManager.hpp"
+#include "ISceneManager.hpp"
+
+#ifdef _WIN32
+#define EXPORT __declspec(dllexport)
```
Comment (Saverio976) [inline include/Api.hpp]: need tabulation

File: include/Api.hpp [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+#else
+#define EXPORT
```
Comment (Saverio976) [inline include/Api.hpp]: need tabulation

File: include/Api.hpp [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+#endif
+
+enum class libType { GRAPH, GAME };
```
Comment (Saverio976) [inline include/Api.hpp:19]: peux tu mettre le enum class dans le extern stp ?

File: include/Api.hpp [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+#include "ISceneManager.hpp"
+#include "ISystemManager.hpp"
+
+#ifdef _WIN32
+    #define EXPORT __declspec(dllexport)
+#else
+    #define EXPORT
+#endif
+
+enum class libType { GRAPH, GAME };
```
Comment (Saverio976) [inline include/Api.hpp:19]: "libType" -> "LibType"

**Review by Saverio976 — APPROVED** [review]

**Pull request merged**